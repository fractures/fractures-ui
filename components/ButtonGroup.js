import React, { Component } from "react"

class ButtonGroup extends Component {
	render() {
		return (
			<div className="fr-button-group flex">
				{this.props.children}
				<style jsx global>{`
					.fr-button-group .fr-btn--radius,
					.fr-button-group .radius-max {
						border-radius: 0;
					}

					.fr-button-group > div:not(:first-child) {
						margin: 0 0 0 -1px;
					}

					.fr-button-group .fr-btn:focus:not(.fr-btn--disabled):not(.fr-btn--loading) {
						position: relative;
					}

					.fr-button-group > div:first-child .fr-btn--radius {
						border-top-left-radius: 0.25rem;
						border-bottom-left-radius: 0.25rem;
					}

					.fr-button-group > div:last-child .fr-btn--radius {
						border-top-right-radius: 0.25rem;
						border-bottom-right-radius: 0.25rem;
					}

					.fr-button-group > div:first-child .radius-max {
						border-top-left-radius: 10rem;
						border-bottom-left-radius: 10rem;
					}

					.fr-button-group > div:last-child .radius-max {
						border-top-right-radius: 10rem;
						border-bottom-right-radius: 10rem;
					}
				`}</style>
			</div>
		)
	}
}

export default ButtonGroup
