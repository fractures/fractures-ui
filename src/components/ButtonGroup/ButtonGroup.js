import PropTypes from "prop-types"
import styled from "styled-components"

const propTypes = {
	children: PropTypes.node.isRequired
}

const ButtonGroup = styled.div`
	display: flex;

	& > * > * {
		border-radius: 0;
	}

	& > *:not(:first-child) *:not(.radius-max) {
		margin: 0 0 0 -1px;
	}

	& > *:first-child *:not(.radius-max) {
		border-top-left-radius: 0.25rem;
		border-bottom-left-radius: 0.25rem;
	}

	& > *:last-child *:not(.radius-max) {
		border-top-right-radius: 0.25rem;
		border-bottom-right-radius: 0.25rem;
	}

	& > *:first-child .radius-max {
		border-top-left-radius: 10rem;
		border-bottom-left-radius: 10rem;
	}

	& > *:last-child .radius-max {
		border-top-right-radius: 10rem;
		border-bottom-right-radius: 10rem;
	}
`

ButtonGroup.propTypes = propTypes

export default ButtonGroup
