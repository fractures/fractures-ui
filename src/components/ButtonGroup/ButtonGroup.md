### Button group

```js noeditor
<Fractures theme="blue" />
```

```js
<ButtonGroup>
	<Button type="secondary" value="Active" isActive={true} />
	<Button type="secondary" value="Second" />
</ButtonGroup>
```
