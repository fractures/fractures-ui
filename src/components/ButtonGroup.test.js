import ButtonGroup from "./ButtonGroup.js"

describe("ButtonGroup", () => {
	const componentRender = <ButtonGroup>Children.</ButtonGroup>
	const component = renderer.create(componentRender)
	const tree = component.toJSON()

	it("Renders correctly", () => {
		expect(tree).toMatchSnapshot()
	})
})
