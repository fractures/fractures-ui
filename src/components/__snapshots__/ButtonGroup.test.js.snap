// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup Renders correctly 1`] = `
<div
  className="jsx-2292800584 fr-button-group flex"
>
  Children.
</div>
`;
